os/board/rtl8730e/src/component/wifi/inic/inic_ipc_host_api.c

```diff
@@ -323,15 +334,36 @@ static void inic_ipc_print_int_hdl(VOID *Data, u32 IrqStatus, u32 ChanNum)
 	(void) IrqStatus;
 	(void) ChanNum;
 
+	static km4log_msg_t message_event = { 0 };
+
+	/* receive a log message over IPC */
 	PIPC_MSG_STRUCT ipc_recv_msg = (PIPC_MSG_STRUCT)ipc_get_message(IPC_NP_TO_AP, IPC_N2A_NP_LOG_CHN);
 	char *tmp_buffer = (char *)ipc_recv_msg->msg;
 	DCache_Invalidate((u32)tmp_buffer, ipc_recv_msg->msg_len);
-	
-	/* Print out buffer */
-	lldbg_noarg("%s",tmp_buffer);
 
+	/* pick a buffer to fill. for rapid processing, assume that the higher priority task has already consumed the buffer, otherwise overwrite it */
+	strncpy((char *)g_inic_ipc_logging_buf[g_inic_ipc_logging_buf_ctr], tmp_buffer, ipc_recv_msg->msg_len);
+
+	/* fill message struct */
+	message_event.buffer = (void *)g_inic_ipc_logging_buf[g_inic_ipc_logging_buf_ctr];
+	message_event.buffer_len = CONFIG_KM4_MAX_BUFFER_SIZE;
+
+	/* use mq_send via osif api directly in ISR instead of semaphore-based */
+	if (g_km4_log_queue == NULL || (!osif_msg_send(g_km4_log_queue, &message_event, 0))) {
+		/* mixlog queue handle was invalid, or sending to queue failed, clear the memory here. */
+		DBG_8195A("queue hndl is null or send failed\n");
+
+		/* set the first byte to null to cause string to print empty in case this buffer slot is accidentally reused */
+		g_inic_ipc_logging_buf[g_inic_ipc_logging_buf_ctr][0] = 0;
+		goto NOTIFY_MSG;
+	}
+
+	/* increment to next buffer */
+	g_inic_ipc_logging_buf_ctr = (g_inic_ipc_logging_buf_ctr + 1) % CONFIG_KM4_MAX_QUEUE_SIZE;
```

Comment:
In a situation where the buffer is full and if we try to overwrite the already sent buffer with mqueue, it can cause log order mixing and break the log.

Please add a processing logic for the case of a full buffer.

Reply:
1. When buffer is full, next log should be dropped.
2. When buffer is full, we should use buffer is full using some log. 

@lhenry-realtek Please verify two things above.